extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,241 @@
+                        final StreamsQuery newQuery = new StreamsQuery(entry.getQueryId(), entry.getSparql().get(),
+                                entry.getIsActive().or(false));
+                        runQuery(newQuery);
+                        queriesCache.put(entry.getQueryId(), newQuery);
+                        LOG.trace("Created new query: " + newQuery.toString());
+                    }
+                    break;
+                case DELETE:
+                    LOG.debug("delete query event.");
+                    final StreamsQuery query = queriesCache.get(entry.getQueryId());
+                    if (query != null) {
+                        stopQuery(query.getQueryId());
+                        queriesCache.remove(query.getQueryId());
+                        LOG.trace("Deleted query: " + query.toString());
+                    } else {
+                        LOG.debug("Delete requested a query that does not exist yet.");
+                    }
+                    break;
+                case UPDATE:
+                    LOG.debug("update query event.");
+                    final StreamsQuery updateQuery = queriesCache.get(entry.getQueryId());
+                    if (updateQuery == null) {
+                        LOG.debug("Query: " + entry.getQueryId() + " does not exist yet, cannot perform update.");
+                    } else {
+                        // if the query is currently inactive, and is updated to
+                        // be active, turn on.
+                        if (!updateQuery.isActive() && entry.getIsActive().or(false)) {
+                            final StreamsQuery newQuery = new StreamsQuery(entry.getQueryId(), entry.getSparql().get(),
+                                    entry.getIsActive().or(false));
+                            runQuery(newQuery);
+                            LOG.trace("Starting query: " + newQuery.toString());
+                        } else if (!updateQuery.isActive() && !entry.getIsActive().or(true)) {
+                            // if the query is running and the update turns it
+                            // off, turn off.
+                            stopQuery(updateQuery.getQueryId());
+                            LOG.trace("Stopping query: " + updateQuery.toString());
+                        } else {
+                            LOG.debug("The query is either already running and "
+                                    + "updated to turn on, or is already stopped and is updated to stop running.");
+                        }
+                    }
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Listener used by the {@link QueryManager} to be notified when
+     * {@link QueryChangeLog}s are created or deleted.
+     */
+    private class QueryManagerSourceListener implements SourceListener {
+        @Override
+        public void notifyCreate(final String ryaInstanceName, final QueryChangeLog log) {
```

Comment:
This method needs an info log saying "Discovered new Query Change Log for Rya Instance [instance name here] within source [source identifying information here]."

Reply:
done